linera-execution/src/evm/database.rs

```diff
@@ -264,74 +245,55 @@ where
             .expect("The lock should be possible");
         let mut runtime = self.runtime.lock().expect("The lock should be possible");
         let mut batch = Batch::new();
-        let mut list_new_balances = Vec::new();
         for (address, account) in &self.changes {
             if !account.is_touched() {
                 continue;
             }
-            let val = self.get_contract_address_key(address);
-            if let Some(val) = val {
-                let key_prefix = vec![val, KeyCategory::Storage as u8];
-                let key_info = vec![val, KeyCategory::AccountInfo as u8];
-                let key_state = vec![val, KeyCategory::AccountState as u8];
-                if account.is_selfdestructed() {
-                    batch.delete_key_prefix(key_prefix);
-                    batch.put_key_value(key_info, &AccountInfo::default())?;
-                    batch.put_key_value(key_state, &AccountState::NotExisting)?;
+            let key_prefix = self.get_address_key(KeyCategory::Storage as u8, address);
+            let key_info = self.get_address_key(KeyCategory::AccountInfo as u8, address);
+            let key_state = self.get_address_key(KeyCategory::AccountState as u8, address);
```

Comment:
Would it make sense to eventually use a collections of `View`s here to avoid the boilerplate?